caas/kubernetes/provider/k8swatcher.go

```diff
@@ -25,17 +29,31 @@ type kubernetesNotifyWatcher struct {
 	clock    jujuclock.Clock
 	catacomb catacomb.Catacomb
 
-	out       chan struct{}
-	name      string
-	k8watcher watch.Interface
+	out      chan struct{}
+	name     string
+	informer cache.SharedIndexInformer
 }
 
-func newKubernetesNotifyWatcher(wi watch.Interface, name string, clock jujuclock.Clock) (*kubernetesNotifyWatcher, error) {
+// NewK8sWatcherFunc defines a function which returns a k8s watcher based on the supplied config.
+type NewK8sWatcherFunc func(
+	informer cache.SharedIndexInformer,
+	name string,
+	clock jujuclock.Clock) (KubernetesNotifyWatcher, error)
+
+type WatchEvent string
+
+var (
+	WatchEventAdd    WatchEvent = "add"
+	WatchEventDelete WatchEvent = "delete"
+	WatchEventUpdate WatchEvent = "update"
+)
+
+func NewKubernetesNotifyWatcher(informer cache.SharedIndexInformer, name string, clock jujuclock.Clock) (KubernetesNotifyWatcher, error) {
```

Comment:
this doesn't need to be exported
(and there already an alias in export_test)
same with the strings one